src/main/java/org/jcodec/containers/dpx/DPXReader.java

```diff
@@ -0,0 +1,204 @@
+package org.jcodec.containers.dpx;
+
+import static java.nio.ByteBuffer.allocate;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.ReadableByteChannel;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.jcodec.common.io.IOUtils;
+import org.jcodec.common.io.NIOUtils;
+import org.jcodec.common.io.SeekableByteChannel;
+
+public class DPXReader {
+    private static final int READ_BUFFER_SIZE = 2048 + 1024;
+
+    static final int IMAGEINFO_OFFSET = 768;
+    static final int IMAGESOURCE_OFFSET = 1408;
+    static final int FILM_OFFSET = 1664;
+    static final int TVINFO_OFFSET = 1920;
+
+    public static final int SPDX = 0x53445058;
+    private final ByteBuffer readBuf;
+    private final int magic;
+    private boolean eof;
+
+
+    public DPXReader(SeekableByteChannel ch) throws IOException {
+        this.readBuf = allocate(READ_BUFFER_SIZE);
+        readBuf.order(ByteOrder.BIG_ENDIAN);
+
+        initialRead(ch);
+        this.magic = readBuf.getInt();
+        if (magic != SPDX) {
+            readBuf.order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
can you explain this?
why do you set buffer to LITTLE_ENDIAN when header magic does not match?